<launch>

    <!-- This node will launch frame conversion from vision pose (tf) to mavros pose -->
    <arg name="target_frame_id"     default="/camera_odom_frame" />
    <param name="target_frame_id"   value="$(arg target_frame_id)" />

    <arg name="source_frame_id"     default="/camera_link" />
    <param name="source_frame_id"   value="$(arg source_frame_id)" />

    <param name="output_rate"       value="20" />
      
    <arg name="enable_precland"     default="true" />
    <param name="enable_precland"   value="$(arg enable_precland)" />

    <arg name="precland_target_frame_id"     default="/landing_target" />
    <param name="precland_target_frame_id"   value="$(arg precland_target_frame_id)" />

    <arg name="precland_camera_frame_id"     default="/camera_fisheye2_optical_frame" />
    <param name="precland_camera_frame_id"   value="$(arg precland_camera_frame_id)" />

    <!-- Two steps alignment: 
    1) r,p,y: align current camera frame with default camera frame (x forward, y left, z up)
    2) gamma: align default camera frame's x axis with world y axis
    Frontfacing:
        Forward, USB port to the right (default): r = 0,          p = 0,          y = 0,  gamma = -1.5707963 
        Forward, USB port to the left           : r = 3.1415926,  p = 0,          y = 0,  gamma = -1.5707963 
    Downfacing: you need to tilt the vehicle's nose up a little (not flat) when launch the T265 realsense-ros node, otherwise the initial yaw will be randomized, read here: https://github.com/IntelRealSense/librealsense/issues/4080
    Tilt the vehicle to any other sides and the yaw might not be as stable.
        Downfacing, USB port to the right :       r = 0,          p = -1.5707963, y = 0,  gamma = -1.5707963
        Downfacing, USB port to the left  :       r = 3.1415926,  p = -1.5707963, y = 0,  gamma = -1.5707963
        Downfacing, USB port to the back  :       r = -1.5707963, p = -1.5707963, y = 0,  gamma = -1.5707963          
        Downfacing, USB port to the front :       r = 1.5707963,  p = -1.5707963, y = 0,  gamma = -1.5707963

    Default for this launch file: downfacing, USB port to the right
    -->
    <param name="roll_cam"          value="0" />
    <param name="pitch_cam"         value="-1.5707963" />
    <param name="yaw_cam"           value="0" />
    <param name="gamma_world"       value="-1.5707963" />

    <node pkg="vision_to_mavros" type="vision_to_mavros_node" name="t265_to_mavros" output="screen" >
        <remap from="vision_pose" to="/mavros/vision_pose/pose" />
        <remap from="landing_raw" to="/mavros/landing_target/raw" />
    </node>
</launch>
